tests/kubevirtresource.go

```diff
@@ -0,0 +1,188 @@
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var (
+	KubeVirtDefaultConfig v1.KubeVirtConfiguration
+	originalKV            *v1.KubeVirt
+)
+
+func AdjustKubeVirtResource() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	util.PanicOnError(err)
+
+	kv := util.GetCurrentKv(virtClient)
+	originalKV = kv.DeepCopy()
+
+	KubeVirtDefaultConfig = originalKV.Spec.Configuration
+
+	if !flags.ApplyDefaulte2eConfiguration {
+		return
+	}
+
+	// Rotate very often during the tests to ensure that things are working
+	kv.Spec.CertificateRotationStrategy = v1.KubeVirtCertificateRotateStrategy{SelfSigned: &v1.KubeVirtSelfSignConfiguration{
+		CA: &v1.CertConfig{
+			Duration:    &metav1.Duration{Duration: 20 * time.Minute},
+			RenewBefore: &metav1.Duration{Duration: 12 * time.Minute},
+		},
+		Server: &v1.CertConfig{
+			Duration:    &metav1.Duration{Duration: 14 * time.Minute},
+			RenewBefore: &metav1.Duration{Duration: 10 * time.Minute},
+		},
+	}}
+
+	// match default kubevirt-config testing resource
+	if kv.Spec.Configuration.DeveloperConfiguration == nil {
+		kv.Spec.Configuration.DeveloperConfiguration = &v1.DeveloperConfiguration{}
+	}
+
+	if kv.Spec.Configuration.DeveloperConfiguration.FeatureGates == nil {
+		kv.Spec.Configuration.DeveloperConfiguration.FeatureGates = []string{}
+	}
+	kv.Spec.Configuration.DeveloperConfiguration.FeatureGates = append(kv.Spec.Configuration.DeveloperConfiguration.FeatureGates,
+		virtconfig.CPUManager,
+		virtconfig.LiveMigrationGate,
+		virtconfig.IgnitionGate,
+		virtconfig.SidecarGate,
+		virtconfig.SnapshotGate,
+		virtconfig.HostDiskGate,
+		virtconfig.VirtIOFSGate,
+		virtconfig.HotplugVolumesGate,
+		virtconfig.DownwardMetricsFeatureGate,
+		virtconfig.NUMAFeatureGate,
+		virtconfig.MacvtapGate,
+		virtconfig.ExpandDisksGate,
+		virtconfig.WorkloadEncryptionSEV,
+	)
+	kv.Spec.Configuration.SELinuxLauncherType = "virt_launcher.process"
+
+	if kv.Spec.Configuration.NetworkConfiguration == nil {
+		testDefaultPermitSlirpInterface := true
+
+		kv.Spec.Configuration.NetworkConfiguration = &v1.NetworkConfiguration{
+			PermitSlirpInterface: &testDefaultPermitSlirpInterface,
+		}
+	}
+
+	data, err := json.Marshal(kv.Spec)
+	Expect(err).ToNot(HaveOccurred())
+	patchData := fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, string(data))
+	adjustedKV, err := virtClient.KubeVirt(kv.Namespace).Patch(kv.Name, types.JSONPatchType, []byte(patchData), &metav1.PatchOptions{})
+	util.PanicOnError(err)
+	KubeVirtDefaultConfig = adjustedKV.Spec.Configuration
+	nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	if checks.HasFeature(virtconfig.CPUManager) && len(nodes.Items) > 1 {
+		// CPUManager is not enabled in the control-plane node
+		waitForSchedulableNodeWithCPUManager()
+	}
+}
+
+func waitForSchedulableNodeWithCPUManager() {
```

Comment:
IMO this is more of node util.

Reply:
Yes, but it is used only by the previous function so I am not sure it merits moving to `libnode`.
We could move it in a following PR and make it exported.

Reply:
Fine, be me if it's only used here.